genesyscloud/journey_views/resource_genesyscloud_journey_views_utils.go

```diff
@@ -0,0 +1,253 @@
+			filter.VarType = getStringPointerFromInterface(filterMap["type"])
+			if predicatesSlice, ok := filterMap["predicates"].([]interface{}); ok {
+				predicates := make([]platformclientv2.Journeyviewelementfilterpredicate, len(predicatesSlice))
+				for i, predicate := range predicatesSlice {
+					predicateMap, ok := predicate.(map[string]interface{})
+					if ok {
+						predicates[i] = parseJourneyviewelementfilterpredicate(predicateMap)
+					}
+				}
+				filter.Predicates = &predicates
+			}
+		}
+	}
+	return filter
+}
+
+func parseJourneyviewelementfilterpredicate(predicateMap map[string]interface{}) platformclientv2.Journeyviewelementfilterpredicate {
+	var predicate platformclientv2.Journeyviewelementfilterpredicate
+	predicate.Dimension = getStringPointerFromInterface(predicateMap["dimension"])
+	if valuesSlice, ok := predicateMap["values"].([]interface{}); ok {
+		values := make([]string, len(valuesSlice))
+		for i, value := range valuesSlice {
+			if stringValue, ok := value.(string); ok {
+				values[i] = stringValue
+			}
+		}
+		predicate.Values = &values
+	}
+	predicate.Operator = getStringPointerFromInterface(predicateMap["operator"])
+	predicate.NoValue = getBoolPointerFromInterface(predicateMap["noValue"])
+	return predicate
+}
+
+func parseJourneyViewLink(linkMap map[string]interface{}) platformclientv2.Journeyviewlink {
+	var link platformclientv2.Journeyviewlink
+	link.Id = getStringPointerFromInterface(linkMap["id"])
+	constraintWithinSlice, ok := linkMap["constraint_within"].([]interface{})
+	if ok {
+		link.ConstraintWithin = parseJourneyViewLinkTimeConstraint(constraintWithinSlice)
+	}
+	if !ok {
+		log.Printf("wrong type. correct type %T", constraintWithinSlice)
+	}
+	if constraintAfterSlice, ok := linkMap["constraint_after"].([]interface{}); ok {
+		link.ConstraintAfter = parseJourneyViewLinkTimeConstraint(constraintAfterSlice)
+	}
+	link.EventCountType = getStringPointerFromInterface(linkMap["event_count_type"])
+	if joinAttributesSlice, ok := linkMap["join_attributes"].([]interface{}); ok {
+		joinAttributes := make([]string, len(joinAttributesSlice))
+		for i, attr := range joinAttributesSlice {
+			if stringValue, ok := attr.(string); ok {
+				joinAttributes[i] = stringValue
+			}
+		}
+		link.JoinAttributes = &joinAttributes
+	}
+	return link
+}
+
+func parseJourneyViewLinkTimeConstraint(timeConstraintSlice []interface{}) *platformclientv2.Journeyviewlinktimeconstraint {
+	var timeConstraint platformclientv2.Journeyviewlinktimeconstraint
+	for _, elem := range timeConstraintSlice {
+		timeConstraintMap, ok := elem.(map[string]interface{})
+		if ok {
+			timeConstraint.Unit = getStringPointerFromInterface(timeConstraintMap["unit"])
+			timeConstraint.Value = getIntPointerFromInterface(timeConstraintMap["value"])
+		}
+		if !ok {
+			log.Printf("wrong type 2, correct %T", timeConstraintMap)
+		}
+	}
+	return &timeConstraint
+}
+
+func getStringPointerFromInterface(val interface{}) *string {
```

Comment:
You don't need this function, when you read an interface value you can cast it to a type, we have a built in function in the go sdk that can create a pointer to the string. So an example of this would be 
```
timeConstraint.Unit = platformclientv2.String(timeConstraintMap["unit"].(string))
```

The same goes when working with booleans and integers, cast and then use the corresponding function in the go sdk to create a pointer(If you want to do it in one line)

Reply:
The function helps resolve any nil values but the sdk function will result a runtime panic for any optional value that's not provided

Reply:
Ah ok, that's a good point. We might change the sdk functions in the future to have this functionality but for now we can use your functions